examples/step-a/step-a.cc

```diff
@@ -0,0 +1,1654 @@
+      }
+    hsize_t dims[2];
+    H5Sget_simple_extent_dims(attr_space, dims, NULL);
+
+    std::vector<std::vector<double>> matrix_value(dims[0],
+                                                  std::vector<double>(dims[1]));
+
+    double *hdf5_data = (double *)malloc(dims[0] * dims[1] * sizeof(double));
+    H5Aread(attr, H5T_NATIVE_DOUBLE, hdf5_data);
+    for (unsigned int dim0_idx = 0; dim0_idx < dims[0]; ++dim0_idx)
+      {
+        for (unsigned int dim1_idx = 0; dim1_idx < dims[1]; ++dim1_idx)
+          {
+            matrix_value[dim0_idx][dim1_idx] =
+              hdf5_data[dim0_idx * dims[1] + dim1_idx];
+          }
+      }
+
+    H5Sclose(attr_space);
+    H5Aclose(attr);
+
+    free(hdf5_data);
+
+    return matrix_value;
+  }
+
+  void H5Group::write_int_attr(const std::string attr_name, int value) const
+  {
+    // Code from
+    // https://support.hdfgroup.org/ftp/HDF5/current/src/unpacked/examples/h5_attribute.c
+    hid_t attr;
+    hid_t aid;
+
+    /*
+     * Create scalar attribute.
+     */
+    aid  = H5Screate(H5S_SCALAR);
+    attr = H5Acreate2(hdf5Group,
+                      attr_name.data(),
+                      H5T_NATIVE_INT,
+                      aid,
+                      H5P_DEFAULT,
+                      H5P_DEFAULT);
+
+    /*
+     * Write scalar attribute.
+     */
+    H5Awrite(attr, H5T_NATIVE_INT, &value);
+
+    H5Sclose(aid);
+    H5Aclose(attr);
+  }
+
+  void H5Group::write_double_attr(const std::string attr_name,
+                                  double            value) const
+  {
+    // Code from
+    // https://support.hdfgroup.org/ftp/HDF5/current/src/unpacked/examples/h5_attribute.c
+    hid_t attr;
+    hid_t aid;
+
+    /*
+     * Create scalar attribute.
+     */
+    aid  = H5Screate(H5S_SCALAR);
+    attr = H5Acreate2(hdf5Group,
+                      attr_name.data(),
+                      H5T_NATIVE_DOUBLE,
+                      aid,
+                      H5P_DEFAULT,
+                      H5P_DEFAULT);
+
+    /*
+     * Write scalar attribute.
+     */
+    H5Awrite(attr, H5T_NATIVE_DOUBLE, &value);
+
+    H5Sclose(aid);
+    H5Aclose(attr);
+  }
+} // namespace hdf5
+
+
+namespace elastic_wave
+{
+  using namespace dealii;
+
+
+  // @sect3{ElasticWave class}
+
+  template <int dim, typename ParametersT>
+  class ElasticWave
+  {
+  public:
+    ElasticWave(ParametersT parameters_);
+    ~ElasticWave();
+    void run();
+
+  private:
+    void frequency_sweep();
+    void setup_system();
+    void assemble_system(double omega);
+    void solve();
+    void store_position_data();
+    void store_frequency_step_data(unsigned int frequency_idx);
+    void output_results() const;
+
+    ParametersT parameters;
+
+    Triangulation<dim> triangulation;
+
+
+    QGauss<dim>        quadrature_formula;
+    const unsigned int n_q_points;
+
+    DoFHandler<dim> dof_handler;
+
+    // spacedim is 2*dim because of the real and imaginary parts
+    FESystem<dim> fe;
```

Comment:
no, `spacedim==dim` -- it's just that the number of *components* is twice `dim` :-)

Reply:
Ok :)